velox/functions/lib/Re2Functions.cpp

```diff
@@ -978,7 +988,27 @@ std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern) {
       // Ensures that pattern has a single contiguous stream of wildcard
       // characters.
       if (wildcardStart != -1) {
-        return std::make_pair(PatternKind::kGeneric, 0);
+        if (singleCharacterWildcardCount) {
+          return PatternMetadata{PatternKind::kGeneric, 0};
+        }
+        // Since one stream of wildcard characters is already seen, check for
+        // kOptimizedPattern when the beginning of a second stream of wildcard
+        // characters is encountered and no _ wildcards were seen. This avoids
+        // the use of RE2 functions like FullMatch and Extract to check for
```

Comment:
This comment is cryptic. Maybe you want to say that since the pattern to be matched is a fixed string that doesn't require regular expression matches, you can optimize by avoiding regular expression usage.

Reply:
Thanks, updated the comment.